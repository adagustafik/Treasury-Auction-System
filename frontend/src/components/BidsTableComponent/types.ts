import { FilterType } from "../types";

export interface BidsTableProp {
  filterOnly: FilterType;
  limit?: number;
}
