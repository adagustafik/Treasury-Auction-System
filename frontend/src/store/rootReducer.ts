import { combineReducers } from "@reduxjs/toolkit";
import { auctionsReducer } from "./slices/auctions";
import { userReducer } from "./slices/user";
import { bidsReducer } from "./slices/bids";
import { purchasesReducer } from "./slices/purchases";

export const rootReducer = combineReducers({
  userReducer,
  auctionsReducer,
  purchasesReducer,
  bidsReducer,
});
