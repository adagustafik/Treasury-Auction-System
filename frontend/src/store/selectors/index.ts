export * from "./selectUser";
export * from "./selectAuctions";
export * from "./selectBids";
export * from "./selectPurchases";
