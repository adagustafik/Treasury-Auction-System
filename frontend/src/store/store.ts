import {
  configureStore,
  ThunkAction,
  Action,
  AnyAction,
  ThunkDispatch,
} from "@reduxjs/toolkit";
import thunkMiddleware from "redux-thunk";
import { rootReducer } from "./rootReducer";

export const store = configureStore({
  reducer: rootReducer,
  middleware: [thunkMiddleware],
});

export type AppDispatch = typeof store.dispatch &
  ThunkDispatch<RootState, null, AnyAction>;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
