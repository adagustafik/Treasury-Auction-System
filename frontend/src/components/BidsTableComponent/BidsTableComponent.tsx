import React from "react";
import { Table } from "react-bootstrap";
import { Bid } from "../../api/getBidsRequest";
import {
  dateConversionToString,
  filterCurrent,
  filterHistory,
} from "../../utilities";
import { useAppSelector } from "../../store/hooks";
import { selectBids } from "../../store/selectors";
import { BidsTableProp } from "./types";

export function BidsTableComponent({ filterOnly, limit = 100 }: BidsTableProp) {
  const { bids } = useAppSelector(selectBids);

  let bidsTop: Bid[] = [];

  if (bids.length > 0) {
    const bidsFiltered: Bid[] = bids.filter((bid: Bid) =>
      filterOnly === "history"
        ? filterHistory(bid.auction)
        : filterCurrent(bid.auction)
    );
    bidsTop = bidsFiltered.slice(0, limit);
  }

  return (
    <>
      {bidsTop.length > 0 && (
        <Table striped bordered hover>
          <thead>
            <tr>
              <th>Id</th>
              <th>Auction date</th>
              <th>Closing time</th>
              <th>Issue amount</th>
              <th>Bid id</th>
              <th>Bid date</th>
              <th>Bid type</th>
              <th>Bid amount</th>
              <th>Bid rate</th>
              <th>Yield</th>
              <th>Discount</th>
            </tr>
          </thead>
          <tbody>
            {bidsTop.map((bid: Bid) => (
              <tr key={bid.bidId}>
                <td>{bid.auction.auctionId}</td>
                <td>{dateConversionToString(bid.auction.auctionDate)}</td>
                {bid.type === "comp" && (
                  <td>
                    {dateConversionToString(
                      bid.auction.competitiveBidCloseTime
                    )}
                  </td>
                )}
                {bid.type === "noncomp" && (
                  <td>
                    {dateConversionToString(
                      bid.auction.nonCompetitiveBidCloseTime
                    )}
                  </td>
                )}
                <td>{bid.auction.securityOffered}</td>
                <td>{bid.bidId}</td>
                <td>{dateConversionToString(bid.time)}</td>
                <td>
                  {bid.type === "comp" ? "competitive" : "non-competitive"}
                </td>
                <td>{bid.amount}</td>
                <td>{bid.rate === 0 ? "" : bid.rate}</td>
                <td>{bid.auction.yield}</td>
                <td>{bid.auction.discount}</td>
              </tr>
            ))}
          </tbody>
        </Table>
      )}
      {bidsTop.length === 0 && <p>There were no bids made</p>}
    </>
  );
}
