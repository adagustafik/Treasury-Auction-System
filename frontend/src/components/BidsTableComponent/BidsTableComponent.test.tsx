import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { store } from "../../store/store";
import { BidsTableComponent } from "./BidsTableComponent";

it("renders correctly with filter function prop", () => {
  const registerForm = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <BidsTableComponent filterOnly={"current"} />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(registerForm).toMatchSnapshot();
});
