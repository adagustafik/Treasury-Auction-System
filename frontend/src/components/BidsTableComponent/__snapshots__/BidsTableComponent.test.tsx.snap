// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with filter function prop 1`] = `
<p>
  There were no bids made
</p>
`;
